tests/ObjectPoolTests.cpp

```diff
@@ -0,0 +1,146 @@
+#include "doctestUtils.h"
+
+#include <cesium/omniverse/ObjectPool.h>
+#include <doctest/doctest.h>
+#include <sys/types.h>
+
+#include <algorithm>
+#include <cstdint>
+#include <cstdlib>
+#include <memory>
+#include <queue>
+
+#define MAX_TESTED_POOL_SIZE 1024 // the max size pool to randomly generate
+
+// ObjectPool is a virtual class so we cannot directly instantiate it for
+// testing, and instatiating the classes that implement it (FabricGeometryPool
+// and FabricMaterialPool) requires mocking more complicated classes, so we
+// create a bare-bones class here.
+
+class mockObj {
```

Comment:
Style: in general we avoid abbreviations. `Obj` can be `Object`. Do a pass over the rest of the code and unabbreviate where possible. Some abbreviations are fine like `min` and `max`.